bin/varnishncsa/varnishncsa.c

```diff
@@ -166,15 +166,39 @@ static struct ctx {
 
 enum format_policy {
         FMTPOL_FIRST = 0,
+        FMTPOL_REQ,
+        FMTPOL_RESP,
+        FMTPOL_BEREQ,
+        FMTPOL_BERESP,
         FMTPOL_ALWAYS
 };
 
+#define FMTPOL_REQ_SIDE() *CTX.side == 'c' ? FMTPOL_REQ : FMTPOL_BEREQ
+#define FMTPOL_RESP_SIDE() *CTX.side == 'c' ? FMTPOL_RESP : FMTPOL_BERESP
```

Comment:
Side note since my new suggestion would obsolete the macros: make sure to guard complex expressions with parentheses in macro definitions, to be future-proof.